subsys/net/lib/aws_jobs/src/aws_jobs.c

```diff
@@ -229,7 +229,7 @@ static int publish(struct mqtt_client *const client, const uint8_t *job_id,
 		.message.topic = topic,
 		.message.payload.data = payload_data,
 		.message.payload.len = payload_data_len,
-		.message_id = k_cycle_get_32(),
+		.message_id = .message_id = (uint16_t)(k_cycle_get_32() ? k_cycle_get_32() : 1),
```

Comment:
```suggestion
		.message_id = mqtt_helper_get_next_message_id(),
```

Reply:
Problem is that AWS jobs doesn't use the MQTT helper library.

Reply:
Ok, thanks for informing. Though, could it not happen that the value returned by `k_cycle_get_32()` is changed from the first call to the next, so that even though it is not zero in the check it can be when it is set. In other words, should the counter value be stored in a variable first, e.g. msg_id, so that this becomes: 
```suggestion
		.message_id = .message_id = (uint16_t)(msg_id ? msg_id : 1),
```

Reply:
For sure, I haven't tested this much yet :sweat_smile: Thanks for pointing it out.